core/src/main/java/io/snyk/plugins/artifactory/scanner/PackageValidator.java

```diff
@@ -0,0 +1,110 @@
+package io.snyk.plugins.artifactory.scanner;
+
+import io.snyk.plugins.artifactory.configuration.ConfigurationModule;
+import io.snyk.plugins.artifactory.configuration.PluginConfiguration;
+import io.snyk.sdk.model.Severity;
+import io.snyk.sdk.model.TestResult;
+import org.artifactory.exception.CancelException;
+import org.artifactory.repo.RepoPath;
+import org.artifactory.repo.Repositories;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.snyk.plugins.artifactory.configuration.ArtifactProperty.ISSUE_LICENSES_FORCE_DOWNLOAD;
+import static io.snyk.plugins.artifactory.configuration.ArtifactProperty.ISSUE_VULNERABILITIES_FORCE_DOWNLOAD;
+import static java.lang.String.format;
+
+public class PackageValidator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PackageValidator.class);
+
+  private final ConfigurationModule configurationModule;
+  private final Repositories repositories;
+
+  public PackageValidator(ConfigurationModule configurationModule, Repositories repositories) {
+    this.configurationModule = configurationModule;
+    this.repositories = repositories;
+  }
+
+  public void validate(TestResult testResult, RepoPath repoPath) {
+    validateVulnerabilityIssues(testResult, repoPath);
+    validateLicenseIssues(testResult, repoPath);
+  }
+
+  private void validateVulnerabilityIssues(TestResult testResult, RepoPath repoPath) {
+    final String vulnerabilitiesForceDownloadProperty = ISSUE_VULNERABILITIES_FORCE_DOWNLOAD.propertyKey();
+    final String vulnerabilitiesForceDownload = repositories.getProperty(repoPath, vulnerabilitiesForceDownloadProperty);
+    final boolean forceDownload = "true".equalsIgnoreCase(vulnerabilitiesForceDownload);
+    if (forceDownload) {
+      LOG.debug("Allowing download. Artifact Property \"{}\" is \"true\". {}", vulnerabilitiesForceDownloadProperty, repoPath);
+      return;
+    }
+
+    Severity vulnerabilityThreshold = Severity.of(configurationModule.getPropertyOrDefault(PluginConfiguration.SCANNER_VULNERABILITY_THRESHOLD));
+    if (vulnerabilityThreshold == Severity.LOW) {
```

Comment:
not a blocker, just an improvement idea: maybe will be better to calculate the counts of all vuln severities and then throw a single error

Reply:
Yes, this is the old logic I didn't want to touch here - this PR just moves code around. About to open another PR with logic changes where I do exactly that!

Reply:
...here is the follow-up: https://github.com/snyk/artifactory-snyk-security-plugin/pull/106